### ISSUE TITLE
WCAG 2.1 public comment - sent Dec 28, 2017

### ISSUE THREAD

**Issue by awkawk** [w3c/wcag21#736]
Dear W3C AGWG menbers,
Thank you so much for striving to improve WCAG.

As for the WCAG 2.1 Final Working Draft, please let me send you some
comments as follows. Due to my poor English skill, please forgive me
describing in Japanese.

// Makoto-san, please ask me anytime when you have difficulty in
understanding what I've meant. Thanks in advance! :)

With kindest regards,
Kazuhiko Tsuchiya (@caztcha)

------- (comments start) -----

2.2.6 Accessible Authentication (A)

ユーザー認証の際に、記憶 (recalling) や転記 (transcribing)
に依存しないようにすることが求められていますが、一般的に広く用いられているパスワード認証は、この達成基準に抵触しそうな気がします。

パスワード認証をさせていても、Cookie への保存を妨げていなければ、この達成基準は満たしている、という理解で正しいですか？(その場合、Cookie
の保存期間についての指針があるとよいと思います。)

1.3.4 Identify Common Purpose (AA)

この達成基準は、「7. Common Purposes for User Interface
Components」で挙げられている各種コントロールについて、これらのコントロールの種別をマシンが識別できるようにする、ということでしょうか。(たとえば、Next
/ Previous のボタンがあったとして、button であるというセマンティクスだけではなく、「previous
button」であるというセマンティクスを付すべき、ということでしょうか。)

もしそうであれば、現時点では実現のハードルが高すぎるので、レベルを AAA にするのが妥当と考えます。

1.4.10 Reflow (AA)

ひとつめの「Notes」について。
320 CSS pixels is equivalent to a starting viewport width of 1280 CSS
pixels wide at 400% zoom.
...という例示はこの達成基準の註釈としてはミスリーディングな気がします。もともと 1280 CSS pixels 幅あるウェブコンテンツを
(400%に) ズームした場合、縦横2方向のスクロールが出現するからです。

1.4.11 Graphics Contrast (AA)

文字だけでなく、グラフィックに対してもコントラスト比に関する達成基準が設けられるのは素晴らしいことです。ただ、SC 1.4.3
(文字の場合、4.5:1 のコントラスト比を求めている)
との不整合があるので、その理由が明示的であると、ガイドラインとしてすっきりすると思います。(私自身は、これまで関わったプロジェクトで、アイコンやボタンのコントラストについても
SC 1.4.3 に準じるべきと解釈して、4.5:1 を保持するようにデザインしていました。)

1.4.12 Text Spacing (AA)

Word spacing (単語間の間隔) も達成要件に入っていますが、日本語のように単語間のスペースがない言語については、例外にすべきと考えます。
たとえば日本語の場合、仮に単語間にスペースを入れると、助詞 (particles) や助動詞 (auxiliary verbs)
が分離され、かえって認知的に読みにくくなります。

1.4.13 Content on Hover or Focus (AA)

「Dismissable」について。addditional content を非表示にさせるメカニズムを "without moving
pointer hover or keyboard focus" で実装する、というのがよくわからなかったのですが、これはたとえば Esc
キーを押して追加コンテンツを閉じる、といったインタラクションを意味していますか？

「Hoverable」について。
http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown

で採り上げられている問題 (利用者にとって予期せぬ形で additional content が閉じてしまう)
を解決する主旨だと推察します。であれば、「たとえ利用者がポインタを "近道" で (もとの hover フォーカスが外れる形で)
動かしたとしても」のような記載があるとよいと思いました。

「Persistent」について。hover が外れ次第 additional content
が閉じることをよしとしているように読み取れるのですが、そうなると、
http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown

で指摘されている問題 (同ページの3つめの GIF アニメーションのような、ユーザーにとって不快な挙動)
を引き起こす可能性があり、ややミスリーディングな気がします。

2.5.3 Target Size (AA)

一辺が 22 CSS pixels でよいと書かれていますが、たとえば width=44 px, height=22px
の横長のタッチターゲットが縦に隙間なく配置されている場合、誤タップを引き起こしそうな気がします。例外として「Inline」が挙げられているので、であれば、基本的にタッチターゲットのミニマムサイズは
44 x 44 CSS pixles としてよいのでは？と思います。

iOS Human Interface Guidelines では "Provide ample touch targets for
interactive elements" として 44pt x 44pt をミニマムのタッチターゲットサイズとしています。
https://developer.apple.com/ios/human-interface-guidelines/visual-design/adaptivity-and-layout/

Android の UI ガイドラインでは 48dp x 48dp をミニマムのタッチターゲットサイズとしています。
https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-touch-target-size

**Comment by MakotoUeki** [comment]
**English translation**

**2.2.6 Accessible Authentication (A)**
This success criterion demands not to rely on "recalling" or "transcribing" when user authentication, and I feel that generally-and-widely used "password authentication" seems to contradict this success criterion.

As an single "A" level criterion, if a website with general "password authentication" does not prevent saving the password to cookie, that website should be judged as meeting this success criterion. Is my understanding correct? (If so, it might be better to represent some guidelines of retention period of cookies.)


**1.3.4 Identify Common Purpose (AA)**
I interpreted this success criterion that; for each of the controls listed in "7. Common Purposes for User Interface Components", the type of these controls must be made identifiable by the machine. (For example, given the next / previous buttons, having semantics that these buttons are just "button" is not enough, and we should attach the semantics of "previous" button and "next" button.)

If so, it is very difficult to achieve at this time , so I think that it is reasonable to make this success criterion to level "AAA".


**1.4.10 Reflow (AA)**
As for the first "NOTE"; 
The description of "320 CSS pixels is equivalent to a starting viewport width of 1280 CSS pixels wide at 400% zoom." seems to be misleading as a note for this success criterion. Because when you zoom a web content originally with 1280 CSS pixels to 400% scale, then the two direction (both vertical and horizontal) scrolling should appears for the 320 CSS pixels-wide viewport.


**1.4.11 Graphics Contrast (AA)**
I'm happy that there is a success criterion for contrast ratios for graphics, in addition to SC 1.4.3 and 1.4.6 (contrast ratios for text). But this new success criterion demands contrast ratio of 3:1, lower than the contrast ratio criterion for text (4.5:1). I think that it is good if the reason of this difference is explicit. In my past website design project, I interpreted that the contrast of icons, buttons etc should also conform to SC 1.4.3, and tried hard to keep the ratio of 4.5:1.


**1.4.12 Text Spacing (AA)**
"Word spacing" is included in the list. I understand the importance of this list item, but for languages which don't have spacing between words (such as Japanese), this item should be exempted. In the case of Japanese language, if you put a space between words, "particles" and "auxiliary verbs" are separated, making it cognitively difficult to read.


**1.4.13 Content on Hover or Focus (AA)**
As for "Dismissable";
I don't have confidence that I understand this item correctly, but does this mean interactions such as pressing Esc key to close "additional content" ?

As for "Hoverable";
I interpreted that this item is for solving problems as described in http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown , to prevent "additional content" from closing unexpectedly to the user. If so, I think it is preferable to have a description like "Even if the user takes shortcut pointer movement (in a way that the original hover focus drops off)".

As for "Persistent";
This item seems to allow "additional content" closing as soon as the hover is removed, and I feel it misleading because this doesn't solve problems as described in http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown (unpleasant behavior for users as shown in the 3rd GIF animation).


**2.5.3 Target Size (AA)**
This success criterion says that one side can be 22 CSS pixels. But if there are some "44 px-wide and 22px-high" touch objects and being stacked vertically with no margins, it is likely to cause incorrect taps.

Because this success criterion already has excemption note of "Inline", I think that the minimum size of the touch target can be set as 44 x 44 CSS pixles.

(References)
iOS Human Interface Guidelines says "Provide ample touch targets for interactive elements" and sets the minimum touch target size as 44pt x 44pt.
https://developer.apple.com/ios/human-interface-guidelines/visual-design/adaptivity-and-layout/

Android UI guidelines says that the minimum touch target size is 48dp x 48dp.
https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-touch-target-size

**Comment by awkawk** [comment]
(Official WG Response)
Thank you for your comment. Specific responses are shown inline below:

> 2.2.6 Accessible Authentication (A)
This success criterion demands not to rely on "recalling" or "transcribing" when user authentication, and I feel that generally-and-widely used "password authentication" seems to contradict this success criterion.

>As an single "A" level criterion, if a website with general "password authentication" does not prevent saving the password to cookie, that website should be judged as meeting this success criterion. Is my understanding correct? (If so, it might be better to represent some guidelines of retention period of cookies.)

The Working Group could not reach consensus on this criterion, so it has been removed from the draft and will be considered in a future version.

> 1.3.4 Identify Common Purpose (AA)
I interpreted this success criterion that; for each of the controls listed in "7. Common Purposes for User Interface Components", the type of these controls must be made identifiable by the machine. (For example, given the next / previous buttons, having semantics that these buttons are just "button" is not enough, and we should attach the semantics of "previous" button and "next" button.)

> If so, it is very difficult to achieve at this time , so I think that it is reasonable to make this success criterion to level "AAA".

Your interpretation is correct. We are modifying this criterion to address only the purposes for inputs, and aligning the list to use the HTML 5.2 autofill list. We are keeping this at AA, but believe that it is much more achievable now.

> 1.4.10 Reflow (AA)
As for the first "NOTE";

> The description of "320 CSS pixels is equivalent to a starting viewport width of 1280 CSS pixels wide at 400% zoom." seems to be misleading as a note for this success criterion. Because when you zoom a web content originally with 1280 CSS pixels to 400% scale, then the two direction (both vertical and horizontal) scrolling should appears for the 320 CSS pixels-wide viewport.

The working group specified one dimension, but since the goal is to avoid text running off screen in more than one dimension the way that this SC works is that the author needs to make sure that the content can still be used at a screen width of 320 pixels. The height may vary but since the wrapping generally needs to happen horizontally the test in the SC is based on the width. For vertical text we are also clarifying in a note that the width dimension is treated as height since the text is flowing in a different direction.

> 1.4.11 Graphics Contrast (AA) I'm happy that there is a success criterion for contrast ratios for graphics, in addition to SC 1.4.3 and 1.4.6 (contrast ratios for text). But this new success criterion demands contrast ratio of 3:1, lower than the contrast ratio criterion for text (4.5:1). I think that it is good if the reason of this difference is explicit. In my past website design project, I interpreted that the contrast of icons, buttons etc should also conform to SC 1.4.3, and tried hard to keep the ratio of 4.5:1.

3:1 is the ratio because there are many more possible combinations of adjacent colors than are typically found in text. We have received many comments to this effect and made the ratio 3:1 in response to public and working group input.

> 1.4.12 Text Spacing (AA) "Word spacing" is included in the list. I understand the importance of this list item, but for languages which don't have spacing between words (such as Japanese), this item should be exempted. In the case of Japanese language, if you put a space between words, "particles" and "auxiliary verbs" are separated, making it cognitively difficult to read.

We are aware of this issue and will be creating an exception to address this.

> 1.4.13 Content on Hover or Focus (AA)
As for "Dismissable"; I don't have confidence that I understand this item correctly, but does this mean interactions such as pressing Esc key to close "additional content" ?

> As for "Hoverable"; I interpreted that this item is for solving problems as described in http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown , to prevent "additional content" from closing unexpectedly to the user. If so, I think it is preferable to have a description like "Even if the user takes shortcut pointer movement (in a way that the original hover focus drops off)".

> As for "Persistent";
This item seems to allow "additional content" closing as soon as the hover is removed, and I feel it misleading because this doesn't solve problems as described in http://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown (unpleasant behavior for users as shown in the 3rd GIF animation).

re: dismissible - your interpretation is correct.

re: hoverable - We understand the issue, but cannot identify clear language to address this in a way that is testable. This issue does also affect all users, so it may be that it is a general usability issue that WCAG wouldn't cover. We will mark this issue as "defer" for future consideration.

re: persistent - This seems to be the same issue mentioned above as "hoverable".

> 2.5.3 Target Size (AA)
This success criterion says that one side can be 22 CSS pixels. But if there are some "44 px-wide and 22px-high" touch objects and being stacked vertically with no margins, it is likely to cause incorrect taps.

> Because this success criterion already has excemption note of "Inline", I think that the minimum size of the touch target can be set as 44 x 44 CSS pixles.

It is worth noting that the Apple and Google guidelines do not include text links, and that is a large part of the challenge in this SC. We understand your interest in a 44x44 size, and this and related questions have made it difficult for the WG to reach consensus on the language for 2.5.3, so the SC has been removed from the latest draft.

**Comment by caztcha** [comment]
Thank you so much for your considering my comments, and sharing me the official WG response above. I think that all of the working group decisions seem reasonable and agree with them.
Just one thing, as for the SC2.5.3 (Target Size), I think it is worth that WCAG is mentioning about the touch target size. If possible, it would be nice if the success criterion 2.5.4 would be kept as it is...
